### PR TITLE
[SAZ-369] Contentful Action is adding env to Engineering role multiple times

### PR DESCRIPTION
**Pull request by waisangu**
[SAZ-369](https://infatuation.atlassian.net/browse/SAZ-369)

Updates action/updateRoleEnvironmentAccess to first check if the environment id exists in the Role's policies. 
If it does, log the reason why it was not added.
If it does not, proceed with adding the environment id to the Role's policies.

[SAZ-369]: https://infatuation.atlassian.net/browse/SAZ-369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/action/updateRoleEnvironmentAccess.ts [hunk 1]
```diff
@@ -8,9 +8,27 @@ export default async function ({ space, environment }: { space: Space; environme
   try {
     Logger.log(`Updating the Role ${roleId} to have access on Environment: ${environment.sys.id}.`);
     const role = await space.getRole(roleId);
-    role.policies.push(createAllowPolicyForEnvironment(environment.sys.id));
-    await role.update();
-    Logger.success(`Successfully updated the Role ${roleId} with Environment ${environment.sys.id}.`);
+
+    let isEnvironmentIdInRolePolicy = false;
```
Comment (mxilievski) [inline src/action/updateRoleEnvironmentAccess.ts:12]: can we do something like this? 

```
const hasEnvironmentAccess = role.policies.some(policy =>
  policy.effect === 'allow' &&
  Array.isArray(policy.actions) &&
  policy.actions.includes('access') && // No need for @ts-expect-error if actions can include 'access'
  containsEnvironmentId(policy.constraint, environment.sys.id)
);

```

Comment (mxilievski) [reply src/action/updateRoleEnvironmentAccess.ts:12]: can't we simplify this by just checking for `containsEnvironmentId`? At the end of the day, I don't think we need to check `policy.effect === 'allow' &&
  Array.isArray(policy.actions) &&
  policy.actions.includes('access')`.

Comment (mxilievski) [reply src/action/updateRoleEnvironmentAccess.ts:12]: As a final suggestion, inverting the logic and returning first is a good practice, the final code should look something like this: 
`export default async function({ space, environment }: { space: Space; environment: Environment }) {
  const roleId: string = INPUT_UPDATE_ENVIRONMENT_ACCESS_ROLE_ID!;
  try {
    Logger.log(`Updating the Role ${roleId} to have access on Environment: ${environment.sys.id}.`);
    const role = await space.getRole(roleId);
    const isEnvironmentIdInRolePolicy = role.policies.some(policy => containsEnvironmentId(policy.constraint, environment.sys.id));
    if (isEnvironmentIdInRolePolicy) {
      Logger.log(`The Role ${roleId} has already been given access to Environment ${environment.sys.id}.`);
      return;
    }
    role.policies.push(createAllowPolicyForEnvironment(environment.sys.id));
    await role.update();
    Logger.success(`Successfully updated the Role ${roleId} with Environment ${environment.sys.id}.`);
  } catch (error) {
    Logger.error(`Failed to update the Role ${roleId} with Environment ${environment.sys.id}.`);
    Logger.verbose(error);
  }
}
`

Comment (waisangu) [reply src/action/updateRoleEnvironmentAccess.ts:12]: i included the checks for `policy.effect` and `policy.actions` in the off chance the EE was manually created in Contentful's UI with the same name as the feature branch, but given different `policy.effect` and `policy.actions` values. this scenario will probably never happen, so i would say it is safe to not include. figured to include it and have the reviewer determine if its necessary

Comment (waisangu) [reply src/action/updateRoleEnvironmentAccess.ts:12]: using a one-liner to iterate through the array is a lot cleaner, and i agree the if logic should not start with bang operator. will update to reflect those changes

Comment (waisangu) [reply src/action/updateRoleEnvironmentAccess.ts:12]: update: your suggestion had a `return` in the if statement, but lint rule asks for a padding-line-between-statements and prettier rule asks to delete the line. instead i wrapped everything after the if statement in an else statement.

Comment (mxilievski) [reply src/action/updateRoleEnvironmentAccess.ts]: Why not just add that line that the linter complains about?
Prematurely returning is much cleaner then if/else

Comment (mxilievski) [reply src/action/updateRoleEnvironmentAccess.ts]: Also https://www.samanthaming.com/tidbits/23-no-else-return/

File: src/action/updateRoleEnvironmentAccess.ts [hunk 2]
```diff
@@ -8,6 +8,13 @@ export default async function ({ space, environment }: { space: Space; environme
   try {
     Logger.log(`Updating the Role ${roleId} to have access on Environment: ${environment.sys.id}.`);
     const role = await space.getRole(roleId);
+    const isEnvironmentIdInRolePolicy = role.policies.some((policy) =>
+      containsEnvironmentId(policy.constraint, environment.sys.id),
+    );
+    if (isEnvironmentIdInRolePolicy) {
+      Logger.log(`The Role ${roleId} has already been given access to Environment ${environment.sys.id}.`);
+      return; // eslint-disable-line padding-line-between-statements
```
Comment (mxilievski) [inline src/action/updateRoleEnvironmentAccess.ts:16]: Why disable? Let’s satisfy the linter and add that blank line before the return.

Comment (waisangu) [reply src/action/updateRoleEnvironmentAccess.ts:16]: adding the blank line initially gave me an prettier lint error:
image

but if i backspace several times on that blank line, it fixes what i thought was conflicting lint issues.

**Review by mxilievski — APPROVED** [review]

**Review by vcanellas — APPROVED** [review]

**Pull request merged**